ui/src/components/retreat/home/PraiseWall.js

```diff
@@ -0,0 +1,280 @@
+  const [input, setInput] = useState('');
+  const { user } = props;
+  const handleInputChange = (e) => setInput(e.target.value);
+
+  const [praiseList, setPraiseList] = useState([]);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const { handleSubmit, register } = useForm();
+
+  const cardBgColors = ['#6dced3', '#ffdc82', '#f39371', '#ffdc82', '#6dced3'];
+  const eventCategory = 'CWC 2022';
+
+  const SplitToChunks = (array, parts) => {
+    let result = [];
+    for (let i = parts; i > 0; i--) {
+      result.push(array.splice(0, Math.ceil(array.length / i)));
+    }
+    return result;
+  };
+
+  const GetPraise = async () => {
+    try {
+      const { data, status } = await axios.get('/api/praises/get');
+
+      if (status === 200) {
+        data.sort((a, b) => (a.createdAt < b.createdAt ? 1 : -1));
+        setPraiseList([...data]);
+      } else {
+        throw Error('Something went wrong');
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const PostPraise = async ({ message }) => {
+    try {
+      const { data } = await axios.post('/api/praises/create', {
+        fullName: user.fullName,
+        message: message,
+        category: eventCategory,
+      });
+      setFormSubmitted(true);
+      GetPraise();
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  const MobileView = () => {
+    return (
+      <Center display={{ base: 'flex', lg: 'none' }}>
+        <HStack align="start" spacing="5">
+          {SplitToChunks(
+            praiseList.filter((praise) => praise.category == eventCategory),
+            2
+          ).map((filteredPraise, i) => (
+            <VStack align="start" key={i} w="12em">
+              {filteredPraise.map((praise, index) => (
+                <PraiseCard
+                  message={praise.message}
+                  cardColor={cardBgColors[(i % 3) + (index % 3)]}
+                  key={i}
+                />
+              ))}
+            </VStack>
+          ))}
+        </HStack>
+      </Center>
+    );
+  };
+  const DesktopView = () => {
+    return (
+      <Center display={{ base: 'none', lg: 'flex' }}>
+        <HStack align="start" spacing="5">
+          {SplitToChunks(
+            praiseList.filter((praise) => praise.category == eventCategory),
+            3
+          ).map((filteredPraise, i) => (
+            <VStack align="start" key={i} w="12em">
+              {filteredPraise.map((praise, index) => (
+                <PraiseCard
+                  message={praise.message}
+                  cardColor={cardBgColors[(i % 3) + (index % 3)]}
+                  key={i}
+                />
+              ))}
+            </VStack>
+          ))}
+        </HStack>
+      </Center>
+    );
+  };
+
+  useEffect(() => {
+    GetPraise();
+    const interval = setInterval(() => GetPraise(), 30000);
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
+
+  return (
+    <Container maxW="container.lg">
```

Comment:
for some reason when i go to `with-everything/` the whole page just redirects me back to `/`

Reply:
seems to not be an issue with the newest main retreat branch